libs/googlePayments.js

```diff
@@ -1,49 +1,43 @@
-const nconf = require('nconf');
-const iap = require('in-app-purchase');
-const subscriptions = require('../libs/subscriptions');
-const mobilePayments = require('./mobilePayments');
-const Bluebird = require('bluebird');
+import iap from 'in-app-purchase';
+import bluebird from 'bluebird';
+import { blocks } from './subscriptions.js';
+import { cancelSubscriptionForUser, scheduleNextCheckForUser } from './mobilePayments.js';
 
 const USERS_BATCH = 10;
-const BASE_URL = nconf.get('BASE_URL');
 
-let api = {};
+const api = {};
 
-api.iapValidate = Bluebird.promisify(iap.validate, {context: iap});
+api.iapValidate = bluebird.promisify(iap.validate, { context: iap });
 
 api.processUser = function processUser (habitrpgUsers, user, jobStartDate, nextScheduledCheck) {
-  let plan = subscriptions.blocks[user.purchased.plan.planId];
-  console.log('Processing google sub for ', user._id);
+  const plan = blocks[user.purchased.plan.planId];
 
   if (!plan) {
-    throw new Error(`Plan ${user.purchased.plan.planId} does not exists. User \{user._id}`);
+    throw new Error(`Plan ${user.purchased.plan.planId} does not exists. User ${user._id}`);
   }
 
   const receipt = user.purchased.plan.additionalData;
 
   receipt.data = typeof receipt.data === 'string' ? JSON.parse(receipt.data) : receipt.data;
 
   return api.iapValidate(iap.GOOGLE, user.purchased.plan.additionalData)
-    .then((response) => {
+    .then(response => {
       if (iap.isValidated(response)) {
-        let purchaseDataList = iap.getPurchaseData(response);
-        let subscription = purchaseDataList[0];
+        const purchaseDataList = iap.getPurchaseData(response);
+        const subscription = purchaseDataList[0];
         if (subscription.expirationDate < jobStartDate) {
```

Comment:
ha, the logic is the same but the conditions are inverted, vs Apple... what :D

Reply:
The logic was also inverted. the if case being cancellation here and another check on iOS. I flipped it here, so that both use the same check/logic for consistency